scenario_player/utils/configuration/settings.py

```diff
@@ -89,23 +87,25 @@ def deposit(self) -> bool:
 
         Defaults to True.
         """
-        return self.get("deposit", True)
+        flag = self.dict.get("deposit", True)
+        assert isinstance(flag, bool)
+        return flag
```

Comment:
I'm looking forward to turning some of these properties into plain attributes. E.g. the `deposit` attribute could probably just be a
```python
self.deposit = loaded_definition.get("deposit", True)
```
line in `__init__`.

Reply:
I'm doing that for a follow up already